config/initializers/at_exit.rb

```diff
@@ -0,0 +1,61 @@
+#
+# Graceful shutdown of web workers
```

Comment:
Puma also has [on_worker_shutdown](https://puma.io/puma/Puma/DSL.html#on_worker_shutdown-instance_method). Would that be more appropriate?

Reply:
Well yes, but the docs says "Note: Cluster mode only.". This is useful both for production and development. This works also on Passenger or any other server, when process exits simply cleanup after itself and also all leftovers.